packages/formik/src/hooks/useFormikComputedState.ts

```diff
@@ -0,0 +1,38 @@
+import isEqual from 'react-fast-compare';
+import { useMemo } from 'react';
+import { FormikApi, FormikComputedState, FormikState } from './../types';
+import { useFormikApi } from './useFormikApi';
+
+export const selectComputedState = <Values>(
+  { isFormValid }: FormikApi<Values>,
+  state: Pick<FormikState<Values>, 'initialValues' | 'values' | 'errors'>
+) => {
+  // we do not want to run a deep equals on every render
+  const dirty = useMemo(
+    () => !isEqual(state.initialValues, state.values),
+    [state.initialValues, state.values]
+  );
+
+  const isValid = isFormValid(state.errors, dirty);
+
+  return {
+    dirty,
+    isValid
+  };
+}
+
+const selectStateToCompute = (state: FormikState<any>) => ({
+  errors: state.errors,
+  values: state.values,
+  initialValues: state.initialValues,
+});
+
+export const useFormikApiComputedState = <Values = any>(formikApi: FormikApi<Values>) => {
```

Comment:
consider removing? or replacing with just `useIsValid` `useIsDirty`

Reply:
Agreed.

Reply:
I tried removing this, but then users lose the ability to calculate validity after useFormik as done right in the example app:

https://github.com/formium/formik/pull/3089/files#diff-13e2cca932df8f5229beec94cf2ea0c9f18dbfe117bcdd1c84f97041b216008fR33

In order to get computed state from the API both with or without context, we have to expose something that can calculate it from either place. useFormikApiX was the convention I was using for that purpose.

Reply:
I moved computed state to `api.useComputedState()` so it is accessible from anywhere, and only expose the hooks `useIsValid` and `useIsDirty`.